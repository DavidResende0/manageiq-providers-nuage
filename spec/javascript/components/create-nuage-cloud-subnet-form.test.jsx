import CreateNuageCloudSubnetForm from '../../../app/javascript/components/create-nuage-cloud-subnet-form'
import * as api from "../../../app/javascript/utils/api";

const fetchRouterMock = jest.spyOn(api, 'fetchRouter').mockResolvedValue('SUCCESS');
const dispatchMock = jest.spyOn(DEFAULT_STORE, 'dispatch');

let renderComponent;
let renderComponentFull;

describe('CreateNuageCloudSubnetForm', () => {
  beforeAll(() => {
    renderComponent = () => shallowRedux(<CreateNuageCloudSubnetForm />);
    renderComponentFull = () => mountRedux(<CreateNuageCloudSubnetForm />);
  });

  beforeEach(() => {
    ManageIQ.record.recordId = 123;
  });

  describe('componentDidMount', () => {
    it('router fetch succeeds', () => {
      fetchRouterMock.mockResolvedValue({ ems_id: 111, ems_ref: 222 });
      let component = renderComponent();
      return fetchRouterMock().then(() => {
        component.update();
        expect(fetchRouterMock).toHaveBeenCalledWith(123);
        expect(component.state()).toEqual({emsId: 111, routerRef: 222, loading: false});
        expect(toJson(component)).toMatchSnapshot();
      });
    });

    it('router fetch fails', () => {
      fetchRouterMock.mockRejectedValue({data: {error: { message: 'MSG'}}});
      let component = renderComponent();
      return fetchRouterMock().catch(() => {
        component.update();
        expect(fetchRouterMock).toHaveBeenCalledWith(123);
        expect(component.state()).toEqual({loading: false, error: 'MSG'});
        expect(toJson(component)).toMatchSnapshot();
      });
    });
  });

  describe('redux bindings', () => {
    it('when fully mounted', () => {
      fetchRouterMock.mockResolvedValue('SUCCESS');
      let component = renderComponentFull();
      return fetchRouterMock().then(() => {
        expect(dispatchMock).toHaveBeenCalledWith({type: 'FormButtons.init',     payload: expect.anything()});
        expect(dispatchMock).toHaveBeenCalledWith({type: 'FormButtons.saveable', payload: expect.anything()});
        expect(dispatchMock).toHaveBeenCalledWith({type: 'FormButtons.pristine', payload: expect.anything()});
      });
    });
  });
});
