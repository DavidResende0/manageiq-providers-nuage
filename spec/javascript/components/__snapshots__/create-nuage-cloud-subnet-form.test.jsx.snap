// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNuageCloudSubnetForm componentDidMount router fetch fails 1`] = `
<p>
  MSG
</p>
`;

exports[`CreateNuageCloudSubnetForm componentDidMount router fetch succeeds 1`] = `
<NuageCloudSubnetForm
  loading={false}
  updateFormState={[Function]}
/>
`;
